tests/kafkatest/tests/core/downgrade_test.py

```diff
@@ -21,71 +21,84 @@
-    @parametrize(kafka_version=str(LATEST_1_1), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_1_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_1_0), compression_types=["snappy"])
-    @parametrize(kafka_version=str(LATEST_0_11_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_11_0), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_10_2), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_10_2), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_10_1), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_10_1), compression_types=["gzip"])
-    @parametrize(kafka_version=str(LATEST_0_10_0), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_10_0), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_9), compression_types=["none"], security_protocol="SASL_SSL")
-    @parametrize(kafka_version=str(LATEST_0_9), compression_types=["snappy"])
-    @parametrize(kafka_version=str(LATEST_0_9), compression_types=["lz4"])
-    @parametrize(kafka_version=str(LATEST_0_8_2), compression_types=["none"])
-    @parametrize(kafka_version=str(LATEST_0_8_2), compression_types=["snappy"])
-    def test_upgrade_and_downgrade(self, kafka_version, compression_types, security_protocol="PLAINTEXT"):
+    @parametrize(version=str(LATEST_2_3), compression_types=["none"])
+    @parametrize(version=str(LATEST_2_3), compression_types=["zstd"])
+    @parametrize(version=str(LATEST_2_2), compression_types=["none"])
+    @parametrize(version=str(LATEST_2_2), compression_types=["zstd"])
+    @parametrize(version=str(LATEST_2_1), compression_types=["none"])
+    @parametrize(version=str(LATEST_2_1), compression_types=["lz4"])
+    @parametrize(version=str(LATEST_2_0), compression_types=["none"])
+    @parametrize(version=str(LATEST_2_0), compression_types=["snappy"])
+    @parametrize(version=str(LATEST_1_1), compression_types=["none"])
+    @parametrize(version=str(LATEST_1_1), compression_types=["lz4"])
+    @parametrize(version=str(LATEST_1_0), compression_types=["none"])
+    @parametrize(version=str(LATEST_1_0), compression_types=["snappy"])
+    @parametrize(version=str(LATEST_0_11_0), compression_types=["none"])
+    @parametrize(version=str(LATEST_0_11_0), compression_types=["lz4"])
+    @parametrize(version=str(LATEST_0_10_2), compression_types=["none"])
+    @parametrize(version=str(LATEST_0_10_2), compression_types=["lz4"])
+    @parametrize(version=str(LATEST_0_10_1), compression_types=["none"])
+    @parametrize(version=str(LATEST_0_10_1), compression_types=["gzip"])
+    @parametrize(version=str(LATEST_0_10_0), compression_types=["none"])
+    @parametrize(version=str(LATEST_0_10_0), compression_types=["lz4"])
+    @parametrize(version=str(LATEST_0_9), compression_types=["none"], security_protocol="SASL_SSL")
```

Comment:
Why not PLAINTEXT here as well? Was there something specific to SASL_SSL for this version?

Reply:
Inherited from the upgrade test. It does seem useful to have some tests which validate the auth flows. For each version, I was considering doing one test with PLAINTEXT/no compression and one with SASL_SSL/some compression. How does that sound?

Reply:
If it doesn't add too much to the runtime, I think it would be good to include some more cases like you suggest.

Reply:
Do we want to support all these versions? I'd vote for only testing 1.0 and newer. 1.0 was released roughly 2 years ago.

Reply:
Given that 1.0 was released 2 years ago, I'd even go with 1.1 as the minimum version.

Reply:
Sounds good to me.